File: patch_banner.py
import re

with open("app/src/main/java/com/example/ui/screens/FeaturedPlanBanner.kt", "r") as f:
    content = f.read()

new_imports = """
import androidx.compose.runtime.*
import androidx.compose.ui.window.Dialog
import com.example.data.local.PlanEntity
"""

content = content.replace("import androidx.compose.runtime.Composable", "import androidx.compose.runtime.Composable\n" + new_imports)

new_code = """
@Composable
fun FeaturedPlanBanner(
    plan: PlanEntity,
    onBuyClick: () -> Unit
) {
    var showDetailsDialog by remember { mutableStateOf(false) }

    if (showDetailsDialog) {
        AlertDialog(
            onDismissRequest = { showDetailsDialog = false },
            title = { Text(text = plan.planName, fontWeight = FontWeight.Bold) },
            text = {
                Column {
                    Text(text = "Final Price: ₹${plan.finalPrice}", color = MaterialTheme.colorScheme.primary, fontWeight = FontWeight.Bold)
                    Spacer(modifier = Modifier.height(16.dp))
                    Text(text = "Benefits included:", fontWeight = FontWeight.Bold)
                    Spacer(modifier = Modifier.height(8.dp))
                    val benefits = plan.features.split("|").filter { it.isNotBlank() }
                    benefits.forEach { b ->
                        Text(text = "• $b")
                    }
                    val contents = plan.contents.split("|").filter { it.isNotBlank() }
                    if (contents.isNotEmpty()) {
                        Spacer(modifier = Modifier.height(8.dp))
                        Text(text = "Contents included:", fontWeight = FontWeight.Bold)
                        Spacer(modifier = Modifier.height(8.dp))
                        contents.forEach { c ->
                            Text(text = "• $c")
                        }
                    }
                }
            },
            confirmButton = {
                TextButton(onClick = { showDetailsDialog = false }) {
                    Text("Close")
                }
            }
        )
    }

    Card(
        modifier = Modifier
            .fillMaxWidth()
            .padding(horizontal = 16.dp),
        shape = RoundedCornerShape(16.dp),
        elevation = CardDefaults.cardElevation(defaultElevation = 4.dp)
    ) {
        Box(
            modifier = Modifier
                .fillMaxWidth()
                .background(
                    brush = Brush.horizontalGradient(
                        colors = listOf(
                            MaterialTheme.colorScheme.tertiary,
                            MaterialTheme.colorScheme.primary
                        )
                    )
                )
                .padding(20.dp)
        ) {
            Column {
                Row(verticalAlignment = Alignment.CenterVertically) {
                    Icon(
                        imageVector = Icons.Default.Star,
                        contentDescription = "Featured",
                        tint = MaterialTheme.colorScheme.onPrimary,
                        modifier = Modifier.size(20.dp)
                    )
                    Spacer(modifier = Modifier.width(8.dp))
                    Text(
                        text = "FEATURED PLAN",
                        style = MaterialTheme.typography.labelMedium,
                        fontWeight = FontWeight.Bold,
                        color = MaterialTheme.colorScheme.onPrimary
                    )
                }
                Spacer(modifier = Modifier.height(8.dp))
                Text(
                    text = plan.planName,
                    style = MaterialTheme.typography.titleLarge,
                    fontWeight = FontWeight.ExtraBold,
                    color = MaterialTheme.colorScheme.onPrimary
                )
                Spacer(modifier = Modifier.height(4.dp))
                Text(
                    text = "Only ₹${plan.finalPrice}",
                    style = MaterialTheme.typography.bodyLarge,
                    fontWeight = FontWeight.Medium,
                    color = MaterialTheme.colorScheme.onPrimary.copy(alpha = 0.9f)
                )
                Spacer(modifier = Modifier.height(16.dp))
                Row(verticalAlignment = Alignment.CenterVertically) {
                    Button(
                        onClick = onBuyClick,
                        colors = ButtonDefaults.buttonColors(
                            containerColor = MaterialTheme.colorScheme.onPrimary,
                            contentColor = MaterialTheme.colorScheme.primary
                        ),
                        shape = RoundedCornerShape(8.dp)
                    ) {
                        Text("Buy Now", fontWeight = FontWeight.Bold)
                    }
                    Spacer(modifier = Modifier.width(8.dp))
                    TextButton(
                        onClick = { showDetailsDialog = true },
                        colors = ButtonDefaults.textButtonColors(contentColor = MaterialTheme.colorScheme.onPrimary)
                    ) {
                        Text("More Info", fontWeight = FontWeight.Bold)
                    }
                }
            }
        }
    }
}
"""

content = re.sub(r'@Composable\nfun FeaturedPlanBanner.*', new_code, content, flags=re.DOTALL)

with open("app/src/main/java/com/example/ui/screens/FeaturedPlanBanner.kt", "w") as f:
    f.write(content)

File: patch_banner.kt
        Box(
            modifier = Modifier
                .fillMaxWidth()
                .background(
                    brush = Brush.horizontalGradient(colors = gradientColors),
                    shape = RoundedCornerShape(16.dp)
                )
                .clickable {
                    if (onBannerClick != null) {
                        onBannerClick(banner)
                    } else {
                        onUpgradeClick()
                    }
                }
        ) {
            // Subtle Jaapi pattern in the background
            Canvas(modifier = Modifier.fillMaxSize()) {
                // Draw a few faint Jaapis
                val jaapiPath = androidx.compose.ui.graphics.Path()
                val color = Color.White.copy(alpha = 0.1f)
                
                // Jaapi 1
                jaapiPath.moveTo(size.width * 0.8f, size.height * 0.2f)
                jaapiPath.lineTo(size.width * 0.95f, size.height * 0.5f)
                jaapiPath.lineTo(size.width * 0.65f, size.height * 0.5f)
                jaapiPath.close()
                drawPath(jaapiPath, color)
                drawLine(
                    color = color,
                    start = androidx.compose.ui.geometry.Offset(size.width * 0.65f, size.height * 0.5f),
                    end = androidx.compose.ui.geometry.Offset(size.width * 0.95f, size.height * 0.5f),
                    strokeWidth = 10f
                )
                
                // Jaapi 2
                val jaapiPath2 = androidx.compose.ui.graphics.Path()
                jaapiPath2.moveTo(size.width * 0.1f, size.height * 0.7f)
                jaapiPath2.lineTo(size.width * 0.3f, size.height * 1.0f)
                jaapiPath2.lineTo(size.width * -0.1f, size.height * 1.0f)
                jaapiPath2.close()
                drawPath(jaapiPath2, color)
                
                // Silhouette of Assam map (simplified abstract shape)
                val mapPath = androidx.compose.ui.graphics.Path()
                mapPath.moveTo(size.width * 0.2f, size.height * 0.8f)
                mapPath.quadraticBezierTo(size.width * 0.4f, size.height * 0.9f, size.width * 0.6f, size.height * 0.6f)
                mapPath.quadraticBezierTo(size.width * 0.8f, size.height * 0.5f, size.width * 0.9f, size.height * 0.2f)
                mapPath.quadraticBezierTo(size.width * 0.7f, size.height * 0.3f, size.width * 0.5f, size.height * 0.2f)
                mapPath.quadraticBezierTo(size.width * 0.3f, size.height * 0.3f, size.width * 0.2f, size.height * 0.8f)
                drawPath(mapPath, Color.White.copy(alpha = 0.05f))
                
                // Gamosa border along the bottom
                val dashWidth = 15f
                var x = 0f
                while (x < size.width) {
                    drawLine(
                        color = Color.Red.copy(alpha = 0.7f),
                        start = androidx.compose.ui.geometry.Offset(x, size.height - 4f),
                        end = androidx.compose.ui.geometry.Offset(x + dashWidth, size.height - 4f),
                        strokeWidth = 8f
                    )
                    x += dashWidth + 10f
                }
            }
